runtime/ops/permissions.rs

```diff
@@ -61,14 +69,21 @@ pub fn op_query_permission(
       ))
     }
   };
-  Ok(perm.to_string())
+  Ok(PermissionStatus {
+    state: if perm == PermissionState::GrantedPartial {
+      PermissionState::Granted.to_string()
+    } else {
+      perm.to_string()
+    },
+    partial: perm == PermissionState::GrantedPartial,
+  })
```

Comment:
We could factor this out to `impl From<PermissionState> for PermissionStatus {}`